core-web/libs/sdk/client/src/lib/client/sdk-js-client.ts

```diff
@@ -223,7 +224,17 @@ export class DotCmsClient {
             const url = `${this.config.dotcmsUrl}/api/v1/page/json${formattedPath}${
                 queryParams ? `?${queryParams}` : ''
             }`;
+
             const response = await fetch(url, this.requestOptions);
+            if (!response.ok) {
+                const error = {
+                    status: response.status,
+                    message: ErrorMessages[response.status] || response.statusText
+                };
+
+                console.error(error);
+                throw error;
```

Comment:
Why not `throw new Error`?

Reply:
Also... how this works in the examples?

Reply:
Because new Error() throws a string, and i think the errors should be an object with code and message